test/functional/feature_reindex_readonly.py

```diff
@@ -0,0 +1,63 @@
+#!/usr/bin/env python3
+# Copyright (c) 2023 The Bitcoin Core developers
+# Distributed under the MIT software license, see the accompanying
+# file COPYING or http://www.opensource.org/licenses/mit-license.php.
+"""Test running bitcoind with -reindex from a read-only blockstore
+- Start two nodes, generate blocks, then restart with -reindex after setting blk files to read-only
+"""
+
+import stat
+from test_framework.test_framework import BitcoinTestFramework
+
+
+class BlockstoreReindexTest(BitcoinTestFramework):
+    def set_test_params(self):
+        self.setup_clean_chain = True
+        self.num_nodes = 2
+        self.extra_args = [
+            [],
+            ["-fastprune"]
+        ]
+
+    def skip_test_if_missing_module(self):
+        self.skip_if_root()
+
+    def reindex_readonly(self):
+        self.connect_nodes(0, 1)
+        addr = self.nodes[1].get_deterministic_priv_key().address
```

Comment:
Hm well, that'll certainly be faster but then we can't verify a successful reindex in the new run-as-root path. Size shouldn't be an issue though because no assumptions are made about it, the test calculates how much data it needs to generate. Let me know what you think about this again after the next push

Reply:
ah ya know what I didn't read your comment correctly I thought you were suggesting filling the blocks with invalid junk but that line with `generateblock()` is cool

Reply:
yeah, ive got this too, will push in a few minutes